### DISCUSSION TITLE
🍺🥃주류 리뷰 커뮤니티🍸🍷

### DISCUSSION THREAD

**Discussion by ypd01018** [ondal-house/zu-story#8] [아이디어 선정 안건]
# 기능
## 주요 기능
- [주류 카테고리화](https://www.foodsafetykorea.go.kr/portal/healthyfoodlife/alType.do)
  - 위스키
  - 보드카
  - 브랜드
  - 청주
  - 탁주
  - 소주
- 카테고리 내 제품 리스트화
- 카테고리 내 제품 sort(연령 별 선호도/ 출시연도? / 평점 순/ 리뷰 갯수 순... 등)
- 제품 별 소개
  - 가격
  - 소개
  - 구매 처 별 가격
- 제품 내 CRUD
  - 후기(사진/별점/추천 제조법/어울리는 안주/숙취정도 포함)
  - QNA
- 제품 추가 요청
- 주류관련 데이터 수집, 후기 관련 데이터 수집
- 소셜 로그인
  - 연령 체크
  - 미가입을 통한 악용 반복 CREATE 방지

## 심화 기능
- 주류 카테고리 별 자유게시판 커뮤니티
- 공공 데이터 포탈을 활용한 데이터 수집 및 가공/시각화
- 채팅방

# 장단점

### 아이디어 관련
- 좋아하는 술 관련이라 재밌게 할 수 있을 것 같음(내가)
- 평소 주류 관련 커뮤니티를 찾아봤을 때 만족할만한 웹/앱을 찾을 수 없었음 (대체로 구매 기반)

### 기술/커리어 관련
- 경험이 어느정도 있으므로 러닝커브 down
- 현재 직무 SE/CDS 포지션을 기반으로 하고 싶었던 및 커리어에 도움이 되는 개발 가능
  - 전반적인 이해도를 가진 상태에서 심화 구현 목표 가능
  - 현업을 통한 경험은 있지만 현업에서 할 수 없는 고민 가능
  - 현재 무관 업무를 하고 있는 동료에게 실무에서 배운 지식 전파 가능
- 현업에서 건드릴 수 없는 작업을 해봄으로서 실무에서 전반적인 이해도 상승 (CDS <-> SE)
- FE/BE 직무 공고에서 주로 필요로 하는 JD를 가져갈 수 있음 (커리어에 유용)
  - BE
    - Java/ Kotlin
    - Spring boot
    - ORM (JPA)
    - RDBMS (MySQL)
    - memcached / Redis
    - OOP
    - JUnit
    - Message Queue (심화)
    - MSA

  - FE
    - JavaScript/TypeScript 
    - Babel를 이용한 ES6+
    - React, Vue.JS (웹), React Native (앱)
    - Webpack
    -  Single Page Application
    - Jest, React-testing-Library

  - Common
    - Unit Test
    - Module Test
    - Design Pattern

  - infra
    - docker
    - Jenkins / Travis CI / GitGub Action
    - AWS / GCP / Azure / NCP

## 단점
- 완전 새로운 시도 및 로우레벨부터 구현을 해보고 싶은 인원일 경우 부적합
- 주정뱅이로 보여서 현타올 수 있음
- [좋은 사례가 이미 있다](https://github.com/woowacourse-teams/2021-jujeol-jujeol)
- ui마저 생각한거랑 똑같음 ㅋㅋ 그만큼 대중적이라는 거지~ 잘 된다는거지~

## 장점이자 단점
- 기술 스택이 사실상 거의 정해짐

**Comment by haneeeul** [comment]
시간이 생각보다 적게 걸릴지도,,?

**Reply by ypd01018** [reply]
생각보다 적게 -> 어느정도일까요?
